pkg/segment/writer/segstore.go

```diff
@@ -1037,50 +1036,72 @@ func (ss *SegStore) flushBloomIndex(cname string, bi *BloomIndex) uint64 {
 		return 0
 	}
 
-	defer bffd.Close()
-
-	var buf bytes.Buffer
-	bufWriter := bufio.NewWriter(&buf)
-
 	// there is no accurate way to find exactly how many bytes the write.to is going to write
 	// and we need that number , so that we write it first and then the actual bloom data
 	// hence this messiness to write it to some buffer, get the bytesWritten count and then do
 	// the actual write
```

Comment:
You should also probably change this comment, as we are not writing to a buffer and we are writing directly to a file.

Reply:
addressed in the latest commit.